starboard/nplb/posix_compliance/posix_time_clock_gettime_test.cc

```diff
@@ -0,0 +1,425 @@
+        std::make_tuple(CLOCK_TAI, ClockIsRequired::OPTIONAL),
+        // Monotonic clocks also should increase.
+        std::make_tuple(CLOCK_MONOTONIC, ClockIsRequired::REQUIRED),
+        std::make_tuple(CLOCK_MONOTONIC_RAW, ClockIsRequired::OPTIONAL),
+        std::make_tuple(CLOCK_MONOTONIC_COARSE, ClockIsRequired::OPTIONAL),
+        std::make_tuple(CLOCK_BOOTTIME, ClockIsRequired::OPTIONAL),
+        std::make_tuple(CLOCK_BOOTTIME_ALARM, ClockIsRequired::OPTIONAL)),
+    [](const ::testing::TestParamInfo<std::tuple<clockid_t, ClockIsRequired>>&
+           info) { return GetClockName(std::get<0>(info.param)); });
+
+class MonotonicClock
+    : public ::testing::TestWithParam<std::tuple<clockid_t, ClockIsRequired>> {
+};
+
+TEST_P(MonotonicClock, IsMonotonic) {
+  auto param = GetParam();
+  clockid_t clock_id = std::get<0>(param);
+  ClockIsRequired requirement = std::get<1>(param);
+  std::string clock_name = GetClockName(clock_id);
+  const int kTrials = 1'000;
+
+  struct timespec ts_initial {};
+  errno = 0;
+  int ret_initial = clock_gettime(clock_id, &ts_initial);
+
+  if (ret_initial == -1 && errno == EINVAL) {
+    if (requirement == ClockIsRequired::OPTIONAL) {
+      GTEST_SKIP() << "Optional " << clock_name
+                   << " is not supported on this system.";
+    }
+  }
+  ASSERT_EQ(0, ret_initial)
+      << "Initial clock_gettime(" << clock_name << ") failed. errno: " << errno
+      << " " << strerror(errno);
+  int64_t initial_time = TimespecToMicroseconds(ts_initial);
+
+  for (int trial = 0; trial < kTrials; ++trial) {
+    struct timespec ts_curr {};
+    errno = 0;
+    ASSERT_EQ(0, clock_gettime(clock_id, &ts_curr))
+        << "clock_gettime(" << clock_name << ") in loop failed for trial "
+        << trial << ". errno: " << errno << " " << strerror(errno);
+    int64_t current_time = TimespecToMicroseconds(ts_curr);
+
+    EXPECT_GE(current_time, initial_time)
+        << clock_name
+        << "-based time should be non-decreasing. Previous: " << initial_time
+        << " us, Current: " << current_time << " us on trial " << trial;
+
+    initial_time = current_time;
+  }
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    PosixTimeClockGettimeTests,
+    MonotonicClock,
+    ::testing::Values(
+        std::make_tuple(CLOCK_MONOTONIC, ClockIsRequired::REQUIRED),
+        std::make_tuple(CLOCK_MONOTONIC_RAW, ClockIsRequired::OPTIONAL),
+        std::make_tuple(CLOCK_MONOTONIC_COARSE, ClockIsRequired::OPTIONAL),
+        std::make_tuple(CLOCK_BOOTTIME, ClockIsRequired::OPTIONAL),
+        std::make_tuple(CLOCK_BOOTTIME_ALARM, ClockIsRequired::OPTIONAL),
+        // CPU time clocks are also monotonic.
+        std::make_tuple(CLOCK_PROCESS_CPUTIME_ID, ClockIsRequired::REQUIRED),
+        std::make_tuple(CLOCK_THREAD_CPUTIME_ID, ClockIsRequired::REQUIRED)),
+    [](const ::testing::TestParamInfo<std::tuple<clockid_t, ClockIsRequired>>&
+           info) { return GetClockName(std::get<0>(info.param)); });
+class CpuTimeClock
```

Comment:
nit: add newline above this (right now it seems associated with the INSTANTIATE macro)